components/page-stack/hooks.ts

```diff
@@ -1,6 +1,32 @@
-import { useContext } from "../../deps.ts";
-import { PageStackContext } from './context.ts'
+import { signal } from "../../deps.ts";
+
+export interface PageState {
+  Component: React.ReactNode;
+  isEscapeDisabled: boolean;
+}
+
+const pageStack$ = signal<PageState[]>([]);
 
 export function usePageStack() {
-  return useContext(PageStackContext);
+  return {
+    pageStack$,
+
+    pushPage: (PageComponent: React.ReactNode, isEscapeDisabled = false) => {
+      pageStack$.set((currentStack) =>
+        currentStack.concat({ Component: PageComponent, isEscapeDisabled })
+      );
+    },
+
+    peekPage: () => {
+      return pageStack$.get()[0];
```

Comment:
do we want to use .peek? or rename our method?

Reply:
renamed to `getTopPage`